code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -27,15 +30,22 @@
 // Please dont override this unless you absolutely have to
 /obj/structure/closet/Initialize(mapload)
 	. = ..()
+	icon_closed = "[icon_state]"
+	icon_opened = "[icon_state]_open"
+	if(shared_open_sprite)
+		icon_opened = shared_open_sprite
+	if(is_wardrobe)
+		icon_opened = "generic_open"
+		open_door_sprite = "generic_door"
```

Comment:
Instead of the `is_wardrobe` var, couldn't you just assign the wardrobes these values?

Reply:
I would, but wardrobes aren't all subtypes of one thing IIRC. I could do that but don't want to make more map edits than needed

Reply:
Setting them directly shouldn't need any map edits. For example the boxing glove closet would look like:
```dm
/obj/structure/closet/boxinggloves
	name = "boxing gloves"
	desc = "It's a storage unit for gloves for use in the boxing ring."
	icon_opened = "generic_open"
	open_door_sprite = "generic_door"
```
Without the `is_wardrobe` var.

Reply:
There are a number of cloets not of the wardrobe subtype that need this, I thought it would be better to do it off a var rather than spamming the same two lines all over the place

Reply:
Ah I thought you wanted me to make everything like this a subtype of wardrobe. I see now.